ADB_world/Worldmap_year/OneYearMap.py

```diff
@@ -0,0 +1,163 @@
+#######################################
+# IMPORTS #############################
+#######################################
+
+import plotly.graph_objects as go
+import pandas as pd
+import json
+import os
+import sys
+
+# Get the directory of the current file
+current_directory = os.path.dirname(__file__)
+# Navigate one level up to get the parent directory
+parent_directory = os.path.abspath(os.path.join(current_directory, os.pardir))
+# Add the parent directory to the Python path
+sys.path.append(parent_directory)
+import def_flight_connection  # Import custom flight connection module
+
+# File path for storing flight connection data
+file_path = os.path.join(parent_directory, "Airport Data/flight_connections_year.json")
+
+#######################################
+# Data import #########################
+#######################################
+
+# If the "flight_connections_year" file doesn't exist, generate and save flight connection data
+if not os.path.exists(file_path):
+    all_connections = []  # List to store all flight connections
+
+    # Define the list of months
+    month_list = ["01-January", "02-February", "03-March", "04-April", "05-May", "06-June",
+                  "07-July", "08-August", "09-September", "10-October", "11-November", "12-December"]
+
+    # Generate flight connection data for each month and save it
+    for month in month_list:
+        def_flight_connection.generate_flight_connections_json(month)
+
+    # Process flight connections for January to create initial DataFrame
+    connections_df, _ = def_flight_connection.process_flight_connections("01-January")
+
+    # Iterate over month_list to gather and concatenate data
+    for month in month_list[1:]:
+        connections_data, _ = def_flight_connection.process_flight_connections(month)
+        connections_df = pd.concat([connections_df, connections_data], ignore_index=True)
+
+    # Consolidate connections with the same departure and destination
+    connections_df = connections_df.groupby(["icao_departure", "icao_destination"], as_index=False).agg({
+        "departure_airport_name": "first",
+        "lat_departure": "first",
+        "lon_departure": "first",
+        "lat_destination": "first",
+        "lon_destination": "first",
+        "averageDailyFlights": lambda x: x.sum() / 12  # Divide the sum by 12 for annual average
+    })
+
+    # Convert DataFrame to dictionary and save as JSON file
+    all_connections.append(connections_df.to_dict(orient='records'))
+    with open(file_path, 'w') as f:
+        json.dump(all_connections, f, indent=4)
+    print(f"All connections saved to {file_path}")
+else:
+    print(f"The file '{file_path}' already exists.")
+
+#######################################
+# Plotting ############################
```

Comment:
I would suggest that you move the plotting code to one self-contained file. That file should _only_ take in data and plot it - not generate data first. I believe it would work best to use a dataframe as the only input file. That would simplify things a lot.